lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestResubmitTask.cs

```diff
@@ -0,0 +1,298 @@
+                    .Set(TaskConfiguration.Task, GenericType<ResubmitTask>.Class)
+                    .Set(TaskConfiguration.OnClose, GenericType<ResubmitTask>.Class)
+                    .Set(TaskConfiguration.OnMessage, GenericType<ResubmitTask>.Class)
+                    .Build();
+            }
+        }
+
+        private sealed class ResubmitTask : ITask, IDriverMessageHandler, IObserver<ICloseEvent>
+        {
+            private readonly CountdownEvent _suspendSignal = new CountdownEvent(1);
+
+            [Inject]
+            private ResubmitTask()
+            {
+            }
+
+            public byte[] Call(byte[] memento)
+            {
+                Logger.Log(Level.Info, "Hello in ResubmitTask");
+                _suspendSignal.Wait();
+                return null;
+            }
+
+            public void Dispose()
+            {
+                Logger.Log(Level.Info, "Task is disposed.");
+            }
+
+            /// <summary>
+            /// When receiving closed task event, verify the command from the driver. If it matches expected message, 
+            /// throw exception to close the task. Otherwise, signal the task to return, that would result in test failure 
+            /// as the test expect two failed tasks. 
+            /// </summary>
+            /// <param name="value"></param>
+            public void OnNext(ICloseEvent value)
+            {
+                if (value.Value != null && value.Value.Value != null)
+                {
+                    Logger.Log(Level.Info,
+                        "Closed event received in task:" + Encoding.UTF8.GetString(value.Value.Value));
+                    if (Encoding.UTF8.GetString(value.Value.Value).Equals(KillTaskCommandFromDriver))
+                    {
+                        throw new Exception(TaskKilledByDriver);
+                    }
+                    Logger.Log(Level.Error, UnExpectedCloseMessage);
+                    _suspendSignal.Signal();
+                }
+            }
+
+            /// <summary>
+            /// Expect the message from driver. If the message is the same as what is sent from driver, signal the task to properly return
+            /// Otherwise, throw exception which would cause an unexpected failed task therefore failed test verification. 
+            /// </summary>
+            /// <param name="value"></param>
+            public void Handle(IDriverMessage value)
+            {
+                var message = ByteUtilities.ByteArraysToString(value.Message.Value);
+                Logger.Log(Level.Info, "Complete task message received in task:" + message);
+                if (message.Equals(CompleteTaskCommandFromDriver))
+                {
+                    _suspendSignal.Signal();
+                }
+                else
+                {
+                    Logger.Log(Level.Error, UnExpectedCompleteMessage);
+                    throw new Exception(UnExpectedCompleteMessage);                    
```

Comment:
It might make sense to introduce new exception classes for this testing. On the Java side, we have `DriverSideTestFailure` and `EvaluatorSideTestFailure` to capture test failure on the Driver and Evaluator side. That way, we can disambiguate between test failures and other exceptions, especially those that are expected.